src/Swarm/Game/Step.hs

```diff
@@ -249,16 +259,28 @@ randomName = do
 -- Debugging
 ------------------------------------------------------------
 
--- | For debugging only. Print some text via the robot's log.
-traceLog :: (Has (State GameState) sig m, Has (State Robot) sig m) => Text -> m ()
-traceLog msg = do
+-- | Create a log entry given current robot and game time in ticks noting whether it has been said.
+--
+--   This is the more generic version used both for (recorded) said messages and normal logs.
+createLogEntry :: (Has (State GameState) sig m, Has (State Robot) sig m) => Bool -> Text -> m LogEntry
+createLogEntry isSaid msg = do
+  rid <- use robotID
   rn <- use robotName
   time <- use ticks
-  robotLog %= (Seq.|> LogEntry msg rn time)
+  loc <- use robotLocation
+  pure $ LogEntry time isSaid rn rid loc msg
+
+-- | Print some text via the robot's log.
```

Comment:
Why do we need the Bool argument? Can we add a comment explaining what it is for?

Reply:
Thanks for a reminder, I think a small enumeration `data MessageSource = Said | Logged | ErrorTrace` would work best here and I could colour the errors red. :fire:

Reply:
Looks pretty! :art: 
![image](https://user-images.githubusercontent.com/44544735/182708483-18e24a01-531b-4449-bd10-058fbbf214d4.png)